docs_src/design/ucr/0003-Extending-Device-Data.md

```diff
@@ -0,0 +1,106 @@
+
+What is needed is a common means of showing the relationship between these added Resources and their
+original south-bound Device; that is, to indicate that these Resources "extend" the original Device.
+
+It is desirable that the means of conveying this information become standardized for those EdgeX microservices
+which provide and use it, hence proposing here that there be a common EdgeX way defined to do this.
+
+### Target Users
+- Product Developers
+- Device Owner
+- Device User
+- Device Maintainer
+- Cloud User
+- Service Provider
+- Software Integrator
+
+### Description
+Picture the extremely simple case of a south-bound sensor device that just measures Temperature and Humidity and provides these as Device Resources. If we then add analytic and north-bound microservices:
+- A Trending service that has Device Resources to indicate that Temperature and Humidity are trended for, eg, Minimum, Average, and Maximum over a 1 hour trend interval.
+- An Alarming service that has Device Resources to describe the Alarm Rules used to monitor Temperature and Humidity, plus a device-level InAlarm status.
+- A Cloud service that reports not just the Temperature and Humidity but also their Trend configuration and Alarm Rule Resources. 
+In addition, the Cloud service adds its own Resources to direct the Cadence with which this Device's data is reported.
+
+Now scale this up to 100 such Temperature/Humidity sensors, and it grows difficult to match all of the added
+Resources to their original sensor data. And add the requirement that all these resources must be able to be seen 
+and managed locally via REST or Message Bus, and potentially from north-bound services like Modbus/TCP, and from 
+the Cloud (because everybody wants to control everything from the Cloud). 
+
+Furthermore, from the end user's point of view, the Trend configuration, Alarm Rules, and Cloud Cadence that are added for a given Device are all seen as aspects of the Temperature/Humidity Device, as is common 
+for Digital Twin representations, and not as separated, free-standing entities. 
+So there must be some means to relate the extended Device Resources to the original south-bound Device and its
+Device Resources.
+
+
+### Existing solutions
+In Eaton's legacy products, these configuration and status extensions could be added to the channels of the Device in the central data store, exist as channel metadata, or be stored in separate data objects. This was fine and 
+efficient for a self-contained, monolithic solution, but is awkward when dividing the monolith into decoupled
+microservices.
+
+In EdgeX today, Devices and Resources can be added that are not related to the south-bound Device or to each other, 
+except perhaps by well-chosen Labels or Tags.
+
+
+### Requirements
+1. Standard EdgeX means exists to relate configuration and status Resources for Application Services to the
+originating south-bound Device data, on a per-Device basis.
```

Comment:
should be per-Device Class basis? Each device references it Device Profile which describes its list of Device Resources.

Reply:
See my dithering answer above. I think, in the end, "per-Device instance" is right.

Reply:
Please make that explicit. :-)

Reply:
Done